app/src/main/java/com/hedvig/app/feature/embark/passages/previousinsurer/askforprice/AskForPriceInfoActivity.kt

```diff
@@ -0,0 +1,92 @@
+package com.hedvig.app.feature.embark.passages.previousinsurer.askforprice
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.material.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.hedvig.app.R
+import com.hedvig.app.ui.compose.composables.TopAppBarWithBack
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+
+class AskForPriceInfoActivity : ComponentActivity() {
+
+    private val parameter by lazy {
+        intent.getParcelableExtra<AskForPriceInfoParameter>(PARAMETER)
+            ?: throw Error("Programmer error: DATA is null in ${this.javaClass.name}")
+    }
+
+    @ExperimentalAnimationApi
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            HedvigTheme {
+                Scaffold(
+                    topBar = {
+                        TopAppBarWithBack(
+                            onClick = { onBackPressed() },
+                            title = stringResource(R.string.insurely_title)
+                        )
+                    }
+                ) {
+                    AskForPriceScreen(
+                        parameter.selectedInsuranceProvider,
+                        onSkipRetrievePriceInfo = ::finishWithResult,
+                        onNavigateToRetrievePrice = ::startRetrievePriceActivity
+                    )
+                }
+            }
+        }
+    }
+
+    private fun startRetrievePriceActivity() {
```

Comment:
Should we use activity-based screens here? Perhaps we could experiment with Composables as screens for this flow? 🙂

Reply:
Thats a really good idea, I'll look into it

Reply:
Looked into this and we need to target sdk version 31, which causes some problems at the moment. Some of our dependencies like adyen and shake bugs needs to have `exported = true/false` for each activity in their respective manifest. This field is missing, hence we get

`android:exported needs to be explicitly specified for <activity>. Apps targeting Android 12 and higher are required to specify an explicit value for `android:exported` when the corresponding component has an intent filter defined.`

There seems to be some work arounds for this but I'm not sure how stable those are. Should we dig into this more or just wait with upgrading to 31?

Reply:
Wait, how come we need to upgrade to SDK 31? 😮

Reply:
Because of navigation-compose (https://developer.android.com/jetpack/compose/navigation). From my understanding that is the way to go if we want to navigate between composables

Reply:
Is SDK 31 a requirement for using Compose Navigation? 😮

Reply:
From some point on all androidx libraries started building against api 31. [Accompanist](https://github.com/google/accompanist/releases/tag/v0.17.0) release notes is one example. But we can compile against api 31 and still have targetApi as 30. That shouldn't bring problems.